packages/okta-adapter/src/config.ts

```diff
@@ -0,0 +1,712 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { ElemID, CORE_ANNOTATIONS } from '@salto-io/adapter-api'
+import { createMatchingObjectType } from '@salto-io/adapter-utils'
+import { client as clientUtils, config as configUtils, elements } from '@salto-io/adapter-components'
+import { OKTA } from './constants'
+
+const { createClientConfigType } = clientUtils
+const { createUserFetchConfigType, createSwaggerAdapterApiConfigType } = configUtils
+
+export const CLIENT_CONFIG = 'client'
+export const FETCH_CONFIG = 'fetch'
+export const API_DEFINITIONS_CONFIG = 'apiDefinitions'
+
+export type OktaClientConfig = clientUtils.ClientBaseConfig<clientUtils.ClientRateLimitConfig>
+
+export type OktaFetchConfig = configUtils.UserFetchConfig
+export type OktaApiConfig = configUtils.AdapterSwaggerApiConfig & {
+  settingsSwagger?: {
+    typeNameOverrides?: configUtils.TypeNameOverrideConfig[]
+  }
+}
```

Comment:
can remove (we don't have a separate `settingsSwagger`, that probably came from zuora)
```suggestion
export type OktaApiConfig = configUtils.AdapterSwaggerApiConfig
```